ldap/servers/slapd/test-plugins/test_slapi_memberof.c

```diff
@@ -0,0 +1,456 @@
+    for (i = 0; entryScopeExcludeSubtrees && entryScopeExcludeSubtrees[i]; i++);
+    theConfig.sdn_entryScopeExcludeSubtrees = (Slapi_DN **) slapi_ch_calloc(sizeof(Slapi_DN *), i + 1);
+    for (i = 0; entryScopeExcludeSubtrees && entryScopeExcludeSubtrees[i]; i++) {
+        theConfig.sdn_entryScopeExcludeSubtrees[i] = slapi_sdn_new_dn_byval((const char *)entryScopeExcludeSubtrees[i]);
+    }
+    theConfig.maxgroup = maxgroup;
+
+    return SLAPI_PLUGIN_SUCCESS;
+}
+
+static int
+test_slapi_memberof_close(Slapi_PBlock *pb __attribute__((unused)))
+{
+    return SLAPI_PLUGIN_SUCCESS;
+}
+
+/****************************************************
+ * Test Slapi_memberof Extended Operation
+ ***************************************************/
+static int
+test_slapi_memberof_extend_exop(Slapi_PBlock *pb)
+{
+    char *oid = NULL;
+    int ret = SLAPI_PLUGIN_EXTENDED_NOT_HANDLED;
+    int i;
+    int idx = 0;
+    int count;
+    char error_buffer[1024] = {0};
+    int32_t error_buffer_size;
+    Slapi_MemberOfConfig config = {0};
+    Slapi_MemberOfResult groupvals = {0};
+    Slapi_DN *target_sdn;
+    BerElement *req_bere = NULL;
+    struct berval *reqdata = NULL;
+    char *req_dn = NULL;
+    BerElement *respber = NULL;
+    struct berval *respdata = NULL;
+    Slapi_Value *v;
+    struct berval **returned_bervals = NULL;
+    char **returned_array = NULL;
+
+    error_buffer_size = sizeof(error_buffer);
+
+    if (!slapi_plugin_running(pb)) {
+        slapi_log_err(SLAPI_LOG_ERR, TEST_SLAPI_MEMBEROF_PLUGIN_SUBSYSTEM,
+                      "test_slapi_memberof_extend_exop - plugin not started\n");
+        return ret;
+    }
+
+    slapi_log_err(SLAPI_LOG_NOTICE, TEST_SLAPI_MEMBEROF_PLUGIN_SUBSYSTEM,
+                  "--> test_slapi_memberof_extend_exop\n");
+        /* Fetch the request OID */
+    slapi_pblock_get(pb, SLAPI_EXT_OP_REQ_OID, &oid);
+    if (!oid) {
+        slapi_log_err(SLAPI_LOG_ERR, TEST_SLAPI_MEMBEROF_PLUGIN_SUBSYSTEM,
+                      "test_slapi_memberof_extend_exop - Unable to retrieve request OID.\n");
+        return ret;
+    }
+
+    /* decode the exop */
+    slapi_pblock_get(pb, SLAPI_EXT_OP_REQ_VALUE, &reqdata);
+    if (BV_HAS_DATA(reqdata)) {
+        req_bere = ber_init(reqdata);
+        if (req_bere == NULL) {
+            slapi_log_err(SLAPI_LOG_ERR, TEST_SLAPI_MEMBEROF_PLUGIN_SUBSYSTEM,
+                          "test_slapi_memberof_extend_exop - Failed to decode/init the DN from the request.\n");
+            return ret;
+        }
+        ber_scanf(req_bere, "a", &req_dn);
+        ber_free(req_bere, 1);
+        req_bere = NULL;
+    }
+
+    if (req_dn) {
+        slapi_log_err(SLAPI_LOG_NOTICE, TEST_SLAPI_MEMBEROF_PLUGIN_SUBSYSTEM,
+                      "--> Target entry is %s\n", req_dn);
+    } else {
+        slapi_log_err(SLAPI_LOG_NOTICE, TEST_SLAPI_MEMBEROF_PLUGIN_SUBSYSTEM,
+                      "--> Target entry is (fallback) %s\n", theConfig.member_dn);
+    }
```

Comment:
Is this supposed to be debug logging?  I don't think this should be logged by default

Reply:
Oh sorry I see this is for tests, so that is fine and you can ignore my request